index.html

```diff
@@ -287,14 +597,65 @@ <h2>Grammar Extensions</h2>
+        zero or more <a>annotation objects</a>.</li>
+      <li>A <a>node object</a> which is not the value of a non-keyword property,
+        or which is an value of a <a>list object</a>,
+        MUST NOT contain an `@annotation` property.</li>
+      <li>An <a>annotation object</a> is a <a>node object</a> with the restriction
+        that it MUST NOT contain an `@id` <a>entry</a>.</li>
     </ul>
   </section>
 
   <section>
     <h2>Algorithmic Extensions</h2>
 
+    <p>This specification extends the altorithms from [[[JSON-LD11-API]]] [[JSON-LD11-API]]
+      to support the addition of <a>embedded nodes</a> and <a>annotation objects</a>.</p>
+
+    <p>The following algorithm extensions require that the <em>rdfstar</em> flag be set.</p>
+
     <section>
       <h3>Expansion Algorithm</h3>
+
+      <p>Before step 13.4.3.1, and changing 13.4.3.1 to be an "Otherwise, if",
+        add the following steps:</p>
+      <ol>
+        <li>If the <em>rdfstar</em> flag is not set,
+          and <var>expanded active property</var> is `@annotation`,
+          an <em>invalid annotation</em> error has been detected and processing is aborted.</li>
+        <li>If <var>value</var> is a non-empty <a>map</a> and
+          the <em>rdfstar</em> flag is set,
+          set <var>expanded value</var> to the result of using this algorithm recursively
+          passing <var>active context</var>, <var>active property</var>,
+          <var>value</var> for <var>element</var>, <var>base URL</var>,
+          and the <em>frameExpansion</em> and <em>ordered</em> flags.
+          The resulting expanded value MUST BE a valid <a>embedded node</a>, otherwise,
+          an <em>invalid embedded node</em> error has been detected and processing is aborted.</li>
+      </ol>
+
+      <p>After step 13.4.14 add the following step:
+        If <var>expanded property</var> is `@annotation:
+      </p>
+      <ol>
+        <li>Continue with the next key from <var>element</var> if the <em>rdfstar</em> flag is not set.</li>
+        <li>Set <var>expanded value</var> to the result of using this algorithm recursively
+          passing <var>active context</var>, `@annotation` for <var>active property</var>,
+          <var>value</var> for element, <var>base URL</var>,
+          and the <em>frameExpansion</em> and <em>ordered</em> flags.</li>
+      </ol>
+
+      <p>Step 15.1 extends the set of allowed <a>entries</a> to include `@annotation`.</p>
+
+      <p>Before <a data-cite="JSON-LD11-API#alg-expand-only-language">step 18</a> add the following step:
+        If <var>result</var> contains the <a>entry</a> `@annotation`
+        all associated values MUST BE <a>annotation objects</a>, otherwise,
+        an <em>invalid annotation</em> error has been detected and processing is aborted.
+        Additionally, if there is no <var>active property</var>,
+        or <var>expanded active property</var> is `@graph` or `@included`,
```

Comment:
```suggestion
        or <var>active property</var> is `@graph` or `@included`,
```
*expanded active property* again? Am I missing something?